sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -208,22 +206,33 @@ object DataType {
   }
 
   // NOTE: Map fields must be sorted in alphabetical order to keep consistent with the Python side.
-  private[sql] def parseDataType(json: JValue): DataType = json match {
+  private[sql] def parseDataType(
+      json: JValue,
+      fieldPath: String = "",
```

Comment:
single string? How do we resolve ambiguity with dots in names?

Reply:
there can be no ambiguity with the dots in the names, string or a list doesn't make any difference